src/CMakeLists.txt

```diff
@@ -6,6 +6,10 @@ cmake_minimum_required(VERSION 3.10)
 
 project(esp)
 
+if(CMAKE_SYSTEM_NAME MATCHES "Emscripten")
+  set(EMSCRIPTEN 1)
+endif()
```

Comment:
This isn't needed anymore, was fixed with https://github.com/mosra/toolchains/commit/e33a22f2400b29b8d9492f8686fc00ed2c309a1a and that's pulled in with latest Corrade I think.

OTOH I would strongly advise to use `CORRADE_TARGET_EMSCRIPTEN` or `MAGNUM_TARGET_WEBGL` instead of `if(EMSCRIPTEN)` in the CMakeLists, because those have a clearly defined semantics and are consistent with other platform defines.

Reply:
The one issue I ran into using CORRADE_TARGET_EMSCRIPTEN is that it is not available until the add_subdirectory of corrade which does not occur until near the end of src/cmake/dependencies.cmake.

The habitat-sim build seems to be primarily using UNIX, APPLE, and WIN32 so I was also trying to be consistent with that.

I will move sourcing of corrade cmake to earlier in the build. Maybe into the top-level CMakeLists.txt like in magnum CMakeLists.txt does:

# Find Corrade first so we can check on the target
find_package(Corrade REQUIRED Utility)

That way we can follow a style similar to magnum and use CORRADE_TARGET_* consistently and in place of UNIX, APPLE, etc.

Reply:
Right, sorry, forgot about this "chicken and egg" problem :) I'm not sure if moving `add_subdirectory(corrade)` earlier is feasible, it also needs to keep working with `USE_SYSTEM_MAGNUM`.

The only place where `CORRADE_TARGET_EMSCRIPTEN` is not defined yet is until `add_subdirectory(corrade)` is called in `dependencies.cmake`, right? Then I think using `EMSCRIPTEN` in that one file until the `add_subdirectory()` could be okay. But feel free to do whatever seems clearer to you.

In any case, the above piece of code is not needed anymore as it's set by the toolchain now.